staging/src/k8s.io/apimachinery/pkg/labels/selector_test.go

```diff
@@ -17,13 +17,19 @@ limitations under the License.
 package labels
 
 import (
-	"fmt"
 	"reflect"
 	"strings"
 	"testing"
 
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	"k8s.io/apimachinery/pkg/selection"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var (
+	ignoreBadValueDetail = cmpopts.IgnoreFields(field.Error{}, "BadValue", "Detail")
```

Comment:
Only ignore Detail.
Since the values are now simple types, it should be easy to test them in the unit tests.